Added websocket, javascript, css, and templating

kr0wned opened the pull request:
Changes to finalize websocket, javascript, css, and html templating.

Commit:
Added websocket, javascript, css, and templating

trixtur committed:
🎨 Code improvements

Restored original functionality of the web sockets and used buttons to call endpoints;

trixtur committed:
🎨 Removed class from backend and cleaned up code

More code cleaning and there is no need to set structs to empty string.

trixtur committed:
🐛 Fixed issue with taco not spinning

Taco should display but only spin when the start button is pressed.

trixtur reviewed (approved):
I made a couple of changes. If they look good to you, go ahead and merge them.

kr0wned commented:
I like it, I am willing to merge, but I thought we were going to put the front-end in it's own repo.

trixtur committed:
🐛 Fixed issue with reload displaying json

Reload would display json under the template content. I removed this but perhaps we should write some code to check if the total_count is not 0 and display it.

trixtur committed:
🐛 Fixed issue with pressing start mutliple times

Spamming start will now not restart the spinner and refreshing the page if it stops from a client will show the result.

Commit:
Fixes to multiple users starting and ending.

Commit:
Merge branch 'add-ws-js-css-template' of https://github.com/PyrousNET/tacospin-server into add-ws-js-css-template

trixtur committed:
🐛 Fixed issue with websocket on complete

When End is pressed we need to send a final message to the web socket.

Commit:
Cleaned up duplicate code
